### PR TITLE
feat(heading): add component Heading

### PR DESCRIPTION
**Pull request by tajo**
This add a new component Heading (based on the [proposal](https://github.com/uber-web/baseui/blob/master/proposals/heading-component.md))

It also removes the documentation of `baseui/typography/heading` (silent deprecation for now).

### CODE REVIEW

File: documentation-site/static/examples/heading/basic.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+import {Heading, HeadingLevel} from 'baseui/heading';
+
+export default () => (
+  <HeadingLevel>
+    <Heading>Base Web [L1]</Heading>
```
Comment (gergelyke) [inline documentation-site/static/examples/heading/basic.js]: should we extend this example with a paragraph of Lorem ipsum text? To make it super clear how it works

**Comment by gergelyke** [comment]
if we are deprecating `baseui/typography/heading`, should we add warning messages that users should start using the new component?

**Comment by tajo** [comment]
> if we are deprecating `baseui/typography/heading`, should we add warning messages that users should start using the new component?

I thought we could use even a slower approach for this and start with a silent deprecation aka removing the documentation, then a warning in v8 and removal in v9. It's a tiny component but it can be annoying to make the switch. Also, we already have a lot of changes coming with v7 so I don't want to add more pain points into it.

File: documentation-site/static/examples/heading/basic.js (hunk 1 above)
Comment (tajo) [reply documentation-site/static/examples/heading/basic.js]: Added.

**Comment by tajo** [comment]
> lgtm - should we also add a screener test?

Added.

**Pull request merged**